pkg/statistics/handle/storage/read.go

```diff
@@ -246,18 +275,8 @@ func indexStatsFromStorage(sctx sessionctx.Context, row chunk.Row, table *statis
 			!loadAll &&
 			config.GetGlobalConfig().Performance.LiteInitStats
 		if notNeedLoad {
-			idx = &statistics.Index{
-				Histogram:  *statistics.NewHistogram(histID, distinct, nullCount, histVer, types.NewFieldType(mysql.TypeBlob), 0, 0),
-				StatsVer:   statsVer,
-				Info:       idxInfo,
-				Flag:       flag,
-				PhysicalID: table.PhysicalID,
-			}
-			if idx.IsAnalyzed() {
-				idx.StatsLoadedStatus = statistics.NewStatsAllEvictedStatus()
-			}
-			lastAnalyzePos.Copy(&idx.LastAnalyzePos)
-			break
+			delete(table.Indices, histID)
+			return nil
```

Comment:
I think this is a dangerous change.
Before, we never proactively kick stats out of the stats cache. The only possibility that would happen is eviction caused by memory insufficiency.
My suggestion is that, if this column already has metadata, we just keep it with metadata. And only let the LRU handle the eviction.